rust/otap-dataflow/crates/config/src/pipeline/service/telemetry.rs

```diff
@@ -50,6 +49,162 @@ fn default_reporting_interval() -> Duration {
+                }
+
+                // Determine type from first element and convert
+                match &values[0] {
+                    AttributeValue::Bool(_) => {
+                        let bools: Result<Vec<bool>, _> = values
+                            .into_iter()
+                            .map(|v| match v {
+                                AttributeValue::Bool(b) => Ok(b),
+                                _ => Err(de::Error::custom("expected all elements to be booleans")),
+                            })
+                            .collect();
+                        Ok(AttributeValue::Array(AttributeValueArray::Bool(bools?)))
+                    }
+                    AttributeValue::I64(_) => {
+                        let ints: Result<Vec<i64>, _> = values
+                            .into_iter()
+                            .map(|v| match v {
+                                AttributeValue::I64(i) => Ok(i),
+                                _ => Err(de::Error::custom("expected all elements to be integers")),
+                            })
+                            .collect();
+                        Ok(AttributeValue::Array(AttributeValueArray::I64(ints?)))
+                    }
+                    AttributeValue::F64(_) => {
+                        let floats: Result<Vec<f64>, _> = values
+                            .into_iter()
+                            .map(|v| match v {
+                                AttributeValue::F64(f) => Ok(f),
+                                _ => Err(de::Error::custom("expected all elements to be floats")),
+                            })
+                            .collect();
+                        Ok(AttributeValue::Array(AttributeValueArray::F64(floats?)))
+                    }
+                    AttributeValue::String(_) => {
+                        let strings: Result<Vec<String>, _> = values
+                            .into_iter()
+                            .map(|v| match v {
+                                AttributeValue::String(s) => Ok(s),
+                                _ => Err(de::Error::custom("expected all elements to be strings")),
+                            })
+                            .collect();
+                        Ok(AttributeValue::Array(AttributeValueArray::String(strings?)))
+                    }
+                    AttributeValue::Array(_) => {
+                        Err(de::Error::custom("nested arrays are not supported"))
+                    }
+                }
+            }
+        }
+
+        deserializer.deserialize_any(AttributeValueVisitor)
+    }
+}
+
+/// Array attribute value types for telemetry resource attributes.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, JsonSchema)]
+pub enum AttributeValueArray {
+    /// Array of bools
+    Bool(Vec<bool>),
+    /// Array of integers
+    I64(Vec<i64>),
+    /// Array of floats
+    F64(Vec<f64>),
+    /// Array of strings
+    String(Vec<String>),
```

Comment:
Can we use a Box<Vec<AttributeValue>> ? or are you intentionally limiting this to 1 dimension of string array?

Reply:
That's sort of changing but let's not worry about it. 
https://opentelemetry.io/blog/2025/complex-attribute-types/